doc/guides/adding-form-like-behavior.md

```diff
@@ -0,0 +1,318 @@
+        name={inputProps.name}
+        label={inputProps.label}
+        type={inputProps.type}
+        value={inputProps.value}
+        onChange={inputProps.onChange}
+        InputLabelProps={{
+          shrink: true,
+        }}
+        {...inputProps}
+      >
+        {children}
+      </TextField>
+    </Grid>
+  )
+}
+```
+
+### `TodoFormModal.tsx`
+
+Now that we have a reusable input component, we can create our modal component. For the purpose of this guide, we will be creating a simplified Todo, only providing the name and importance attributes. With the building blocks provided in these snippets, you will be able to build out a full todo form with inputs for all attributes & relationships. Here's the full code, which we will break down below:
+
+```tsx
+import { useEffect, useState } from "react"
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, Grid } from "@mui/material"
+import type { RecordType } from "@hatchifyjs/react"
+import { hatchedReact } from "../App.js"
+import FormInput from "./FormInput.js"
+import * as Schemas from "../../schemas.js"
+
+// 🛑 `RecordType` should be `HatchifyRecord`?
+export default function TodoFormModal({ todo, open, handleClose }: { todo?: RecordType<typeof Schemas, typeof Schemas.Todo>; open: boolean; handleClose: () => void }) {
+  const [createTodo] = hatchedReact.model.Todo.useCreateOne()
+  const [updateTodo] = hatchedReact.model.Todo.useUpdateOne()
+  const [values, setValues] = useState({
+    name: todo?.name ?? "",
+    importance: todo?.importance ?? "0",
+  })
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target
+    setValues((prev) => ({ ...prev, [name]: value }))
+  }
+
+  const handleSubmit = () => {
+    if (!todo) {
+      createTodo({
+        name: values.name,
+        importance: Number(values.importance),
+      })
+    } else {
+      updateTodo({
+        id: todo.id,
+        name: values.name,
+        importance: Number(values.importance),
+      })
+    }
+    handleClose()
+  }
+
+  useEffect(() => {
+    setValues({
+      name: todo?.name ?? "",
+      importance: (todo?.importance ?? 0).toString(),
+    })
+  }, [open, todo])
+
+  return (
+    <Dialog open={open} onClose={handleClose}>
+      <DialogTitle>
+        {todo ? "Edit" : "Add"} Todo {todo ? `[${todo.id}]` : ""}
+      </DialogTitle>
+      <DialogContent>
+        <Grid container spacing={4}>
+          <FormInput name="name" label="Name" type="text" value={values.name} onChange={onChange} />
+          <FormInput name="importance" label="Importance" type="number" value={values.importance} onChange={onChange} />
+        </Grid>
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={handleClose}>Cancel</Button>
+        <Button onClick={handleSubmit}>{todo ? "Edit" : "Add"}</Button>
+      </DialogActions>
+    </Dialog>
+  )
+}
+```
+
+That is a lot of code we just added, so let's break it down.
```

Comment:
I wonder if it would be better just to put this as inline comments?

Reply:
good idea! i think i can put a lot of the standard-react stuff in comments and then outside the block specifically highlight the hatchify stuff?